src/natcap/invest/delineateit/delineateit.py

```diff
@@ -559,10 +533,25 @@ def snap_points_to_nearest_stream(points_vector_path, stream_raster_path_band,
             last_time = time.time()
 
         source_geometry = point_feature.GetGeometryRef()
+        geom_name = source_geometry.GetGeometryName()
+        geom_count = source_geometry.GetGeometryCount()
+
+        if source_geometry.IsEmpty():
+            LOGGER.warning(
+                f"FID {point_feature.GetFID()} is missing a defined geometry. "
+                "Skipping.")
+            continue
 
         # If the geometry is not a primitive point, just create the new feature
-        # as it is now in the new vector.
-        if source_geometry.GetGeometryName() != 'POINT':
+        # as it is now in the new vector. MULTIPOINT geometries with a single
+        # component point count as primitive points.
+        # OGR's wkbMultiPoint, wkbMultiPointM, wkbMultiPointZM and
+        # wkbMultiPoint25D all use the MULTIPOINT geometry name.
+        if ((geom_name not in ('POINT', 'MULTIPOINT')) or
+                (geom_name == 'MULTIPOINT' and geom_count > 1)):
+            LOGGER.debug(
+                f"FID {point_feature.GetFID()} ({geom_name}, n={geom_count}) "
+                "only primitive points can be snapped to a stream.")
```

Comment:
Do you think this message is mainly meant for developers or for users? If it's for devs, then I think it's great. If it's for users who chose the snapping option but are now going to get back "unsnapped" points, do you think it makes more sense as a `warning`? If so I might also remove the "primitive points" reference and say something like, "MULTIPOINT geometries cannot be snapped".

Reply:
Yep, `warning` makes more sense for sure!  This is intended to provide a user-facing record of which geometries were not snapped (though there is enough information in the message for us as developers to see why it was not snapped).

I also updated the language to clarify that the geometry could not be snapped.  Your point (ha) about multipoint geometries is a little tricky because a part of this change is all about snapping `MULTIPOINT` geometries, but only if they have exactly one component point.  It was tough to try to capture that nuance in a log message, so I simplified it ... what do you think about the updated text?

Patched in 1eb6b29c